source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs

```diff
@@ -147,6 +147,26 @@ public void CancelStop(ChargePeriod chargePeriod)
             _periods.Add(chargePeriod);
         }
 
+        public void UpdatePrices(Instant startDate, Instant? endDate, IReadOnlyList<Point> newPrices)
+        {
+            ArgumentNullException.ThrowIfNull(newPrices);
+            if (newPrices.Count == 0) return;
+            RemoveExistingChargePrices(startDate, endDate);
+            _points.AddRange(newPrices);
+        }
+
+        private void RemoveExistingChargePrices(
+            Instant startDate,
+            Instant? endDate)
+        {
+            ArgumentNullException.ThrowIfNull(endDate);
+            var removePoints = _points.Where(x => x.Time >= startDate && x.Time <= endDate).ToList();
```

Comment:
The latest point that needs to be removed, will have a `Time` that lies before the `endDate`, hence the `=` needs to be removed.

```suggestion
            var removePoints = _points.Where(x => x.Time >= startDate && x.Time < endDate).ToList();
```